hub/api/tensor.py

```diff
@@ -1,26 +1,22 @@
 from hub.core.sample import Sample
 from typing import List, Sequence, Union, Optional, Tuple, Dict
 from hub.util.shape import ShapeInterval
-from hub.core.meta.tensor_meta import TensorMeta
 
 import numpy as np
 
-from hub.core.tensor import (
-    append_tensor,
-    extend_tensor,
-    read_samples_from_tensor,
-    tensor_exists,
-)
+from hub.core.chunk_engine import ChunkEngine, SampleValue
 from hub.core.typing import StorageProvider
+from hub.core.storage import LRUCache
 from hub.util.exceptions import TensorDoesNotExistError, InvalidKeyTypeError
+from hub.core.tensor import tensor_exists
 from hub.core.index import Index
 
 
 class Tensor:
     def __init__(
         self,
         key: str,
-        storage: StorageProvider,
+        storage: LRUCache,
```

Comment:
I think we shouldn't force cache, it could just be a storageprovider

Reply:
@AbhinavTuli the reason for doing this is so that our `Cachable` interface always works properly. I've reconfigured so that memory cache always gets created, even if its size is set to zero. It shouldn't make any major difference, but will deserialize our cachables nicely for us.